shell/src/state/peer_state.rs

```diff
@@ -332,6 +343,13 @@ impl UpdateIsBootstrapped for PeerState {
     }
 }
 
-pub fn tell_peer(msg: Arc<PeerMessageResponse>, peer: &PeerState) {
-    peer.peer_id.peer_ref.tell(SendMessage::new(msg), None);
+pub fn tell_peer(
+    network_channel: NetworkChannelRef,
```

Comment:
@binier 
I guess, it is enought: `network_channel: &NetworkChannelRef` then you dont need to `network_channel.clone()` all the time

Reply:
Fixed.